components/Charts/Compare/CompareCostsChart.client.vue

```diff
@@ -45,28 +53,58 @@ const chartTitle = computed(() => {
 const getSeries = (): Highcharts.SeriesColumnOptions[] => {
   // Take the first scenario's costs as an example to find out what the second-level breakdowns are.
   const secondLevelCostIds = scenarios.value[0].result.data?.costs[0].children?.map(c => c.id) || [];
-  seriesData.value = secondLevelCostIds?.map((costId, index) => ({
-    type: "column",
-    name: appStore.getCostLabel(costId),
-    borderWidth: 1,
-    borderColor: costsChartPalette[index].rgb,
-    zIndex: secondLevelCostIds.length - index, // Ensure that stack segments are in front of each other from top to bottom.
-    data: scenarios.value.map((scenario) => {
-      const subCost = scenario.result.data?.costs[0].children?.find(c => c.id === costId);
-      const dollarAmount = getDollarValueFromCost(subCost);
-      // costAsGdpPercent is calculated here since the national GDP may vary by scenario if the axis is 'country'.
-      const costAsGdpPercent = costAsPercentOfGdp(dollarAmount, scenario.result.data?.gdp);
-      const y = costBasis.value === CostBasis.PercentGDP ? costAsGdpPercent : dollarAmount;
-      const name = appStore.getCostLabel(subCost?.id || "");
+  const allSeries = secondLevelCostIds?.map((costId, index) => {
```

Comment:
this is alot of calculations going on in the component... should probs be extracted out so can be unit tested nicely... can be future work

Reply:
Agree. I’m doing so in a follow-up branch.